src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp

```diff
@@ -749,4 +749,130 @@ TEST_CASE("RotatedSPOs read and write parameters history", "[wavefunction]")
+  //SPOSet default implementation
+  {
+    //First check calling the mw_ APIs for RotatedSPOs, for which the
+    //underlying implementation just calls the underlying SPOSet mw_ API
+    //In the case that the underlying SPOSet doesn't specialize the mw_ API,
+    //the underlying SPOSet will fall back to the default SPOSet mw_, which is
+    //just a loop over the single walker API.
+    auto spo_ptr0 = std::make_unique<DummySPOSetWithoutMW>("no mw 0");
+    RotatedSPOs rot_spo0("rotated0", std::move(spo_ptr0));
+    auto spo_ptr1 = std::make_unique<DummySPOSetWithoutMW>("no mw 1");
+    RotatedSPOs rot_spo1("rotated1", std::move(spo_ptr1));
+
+    RefVectorWithLeader<SPOSet> spo_list(rot_spo0);
+    spo_list.push_back(rot_spo0);
+    spo_list.push_back(rot_spo1);
+
+    const SimulationCell simulation_cell;
+    ParticleSet elec0(simulation_cell);
+    ParticleSet elec1(simulation_cell);
+
+    RefVectorWithLeader<ParticleSet> p_list(elec0);
+    p_list.push_back(elec0);
+    p_list.push_back(elec1);
+
+    SPOSet::ValueVector psi0(3);
+    SPOSet::ValueVector psi1(3);
+    RefVector<SPOSet::ValueVector> psi_v_list;
+    psi_v_list.push_back(psi0);
+    psi_v_list.push_back(psi1);
+
+    rot_spo0.mw_evaluateValue(spo_list, p_list, 0, psi_v_list);
+    for (int iw = 0; iw < spo_list.size(); iw++)
+    {
+      CHECK(psi_v_list[iw].get()[0] == Approx(123));
+      CHECK(psi_v_list[iw].get()[1] == Approx(456));
+      CHECK(psi_v_list[iw].get()[2] == Approx(789));
+    }
+  }
+  {
+    //In the case that the underlying SPOSet DOES have mw_ specializations,
+    //we want to make sure that RotatedSPOs are triggering that appropriately
+    //This will mean that the underlying SPOSets will do the appropriate offloading
+    //To check this, DummySPOSetWithMW has an explicit mw_evaluateValue which sets
+    //different values than what gets set in evaluateValue. By doing this,
+    //we are ensuring that RotatedSPOs->mw_evaluaeValue is calling the specialization
+    //in the underlying SPO and not using the default SPOSet implementation which
+    //loops over single walker APIs (which have different values enforced in
+    // DummySPOSetWithoutMW
+    auto spo_ptr0 = std::make_unique<DummySPOSetWithMW>("mw 0");
+    RotatedSPOs rot_spo0("rotated0", std::move(spo_ptr0));
+    auto spo_ptr1 = std::make_unique<DummySPOSetWithMW>("mw 1");
+    RotatedSPOs rot_spo1("rotated1", std::move(spo_ptr1));
+
+    RefVectorWithLeader<SPOSet> spo_list(rot_spo0);
+    spo_list.push_back(rot_spo0);
+    spo_list.push_back(rot_spo1);
+
+    const SimulationCell simulation_cell;
+    ParticleSet elec0(simulation_cell);
+    ParticleSet elec1(simulation_cell);
+
+    RefVectorWithLeader<ParticleSet> p_list(elec0);
+    p_list.push_back(elec0);
+    p_list.push_back(elec1);
+
+    SPOSet::ValueVector psi0(3);
+    SPOSet::ValueVector psi1(3);
+    RefVector<SPOSet::ValueVector> psi_v_list;
+    psi_v_list.push_back(psi0);
+    psi_v_list.push_back(psi1);
+
```

Comment:
Please add create and lock resource https://github.com/QMCPACK/qmcpack/blob/b24ba10264d520aa090dcc103957dfb0515516cb/src/QMCWaveFunctions/tests/test_einset.cpp#L203

It should kill some coverage warning.

Reply:
done